src/wallet.rs

```diff
@@ -42,19 +108,44 @@ impl WalletVersion {
         &self,
         workchain: i32,
         key_pair: &KeyPair,
+        sub_wallet_id: Option<i32>,
     ) -> Result<BagOfCells, TonCellError> {
-        let mut data_builder = CellBuilder::new();
-        data_builder
-            .store_u32(32, 0)?
-            // seqno
-            .store_u32(32, 698983191 + workchain as u32)?
-            //wallet_id
-            .store_slice(key_pair.public_key.as_slice())?; // public key
-        if *self == WalletVersion::V4R2 {
-            data_builder.store_bit(false)?;
-            // empty plugin dict
-        }
-        let data_cell = data_builder.build()?;
+        let wallet_id = sub_wallet_id.unwrap_or(698983191 + workchain);
+        let public_key: [u8; 32] = key_pair
+            .public_key
+            .clone()
+            .try_into()
+            .expect("pubkey is always [u8; 32]");
```

Comment:
We avoid using `expect` as it panics. Returning ` TonCellError::InternalError("Invalid public key size".to_string())` looks more suitable.